### PR TITLE
Add version mismatch warning

### PR DESCRIPTION
**Pull request by vadimkerr**

### CODE REVIEW

File: grafana-plugin/src/containers/DefaultPageLayout/DefaultPageLayout.tsx [hunk 1]
```diff
@@ -88,14 +89,33 @@ const DefaultPageLayout: FC<DefaultPageLayoutProps> = observer((props) => {
...
-        {currentTeam?.limits.show_limits_warning && !getItem(currentTeam.limits.warning_text) && (
-          <Alert
-            className={styles.alert}
-            severity="warning"
-            title={currentTeam?.limits.warning_text}
-            onRemove={getRemoveAlertHandler(currentTeam?.limits.warning_text)}
-          />
+        {store.backendVersion && plugin?.version && store.backendVersion !== plugin?.version && (
+          <Alert className={styles.alert} severity="warning" title={'Version mismatch!'}>
+            Please make sure you have the same versions of Grafana OnCall plugin and Grafana OnCall engine, otherwise
```
Comment (Matvey-Kuk) [inline grafana-plugin/src/containers/DefaultPageLayout/DefaultPageLayout.tsx:94]: of **the** Grafana OnCall

File: engine/apps/api/serializers/organization.py [hunk 2]
```diff
@@ -109,7 +110,26 @@ def get_banner(self, obj):
...
+            text = (
+                "Version mismatch! Please make sure you have the same versions of Grafana OnCall plugin "
+                "and Grafana OnCall engine, "
+                "otherwise there could be issues with your Grafana OnCall installation! "
+                f"Current plugin version: {frontend_version}, current engine version: {backend_version}. "
+                "To install the latest Grafana OnCall plugin version, "
+                "visit the plugin configuration page and click Update. "
+                "To update Grafana OnCall engine, re-pull the grafana/oncall image and restart (or see the docs)."
+            )
+            version_warning = {"period_title": "Version mismatch", "show_limits_warning": True, "warning_text": text}
```
Comment (Matvey-Kuk) [inline engine/apps/api/serializers/organization.py:130]: @vadimkerr are you sure it should be a part of `get_limits` method?

**Review by Matvey-Kuk — CHANGES_REQUESTED** [review]
@vadimkerr I think that it may be confusing to give partial instructions about how to update the system in banners. I propose to make a paragraph here: https://grafana.com/docs/grafana-cloud/oncall/open-source/ called "How to update" where we will point users to different instructions for updating helm, docker-compose etc. And link to this paragraph from the product banners.

File: engine/apps/api/serializers/organization.py (hunk 2 above)
Comment (vadimkerr) [reply engine/apps/api/serializers/organization.py:130]: It shouldn't, but the idea is that this warning should pop up on new backend version and old frontend version. So we need to show a warning without changing the frontend, it's the only place where we can show a warning generated on backend I think. Are there any other places on backend that I forgot about?

**Comment by vadimkerr** [comment]
> @vadimkerr I think that it may be confusing to give partial instructions about how to update the system in banners. I propose to make a paragraph here: https://grafana.com/docs/grafana-cloud/oncall/open-source/ called "How to update" where we will point users to different instructions for updating helm, docker-compose etc. And link to this paragraph from the product banners.

Yes, makes sense, let me add a paragraph on it in the docs and reference it in the warning banner.

File: engine/apps/api/serializers/organization.py (hunk 2 above)
Comment (Konstantinov-Innokentii) [reply engine/apps/api/serializers/organization.py:130]: I think for now it's ok, but we need to refactor it later.

**Review by maskin25 — APPROVED** [review]

**Comment by vadimkerr** [comment]
Here's a PR with docs on OSS update: https://github.com/grafana/oncall/pull/242

**Review by Matvey-Kuk — APPROVED** [review]

**Pull request merged**